osu.Desktop/Admin/ElevatedPrivilegesChecker.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Security.Principal;
+using osu.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Graphics;
+using osu.Game.Overlays;
+using osu.Game.Overlays.Notifications;
+
+namespace osu.Desktop.Admin
+{
+    /// <summary>
+    /// Checks if the game is running with elevated privileges (as admin in Windows, root in Unix) and displays a warning notification if so.
+    /// </summary>
+    public class ElevatedPrivilegesChecker : Component
+    {
+        [Resolved]
+        protected NotificationOverlay Notifications { get; private set; }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            bool elevated = false;
+
+            if (OperatingSystem.IsWindows())
+            {
+                var windowsIdentity = WindowsIdentity.GetCurrent();
+                var windowsPrincipal = new WindowsPrincipal(windowsIdentity);
+                elevated = windowsPrincipal.IsInRole(WindowsBuiltInRole.Administrator);
+            }
+            else if (RuntimeInfo.IsUnix)
+            {
+                elevated = Mono.Unix.Native.Syscall.geteuid() == 0;
+            }
```

Comment:
Move this into a `BackgroundDependencyLoader` `load()` method to allow it to run asynchronously. Not sure on the potential performance cost of these lookups but it's safe to say they may be non-zero.

Also I think you should put this in its own method with a `bool` return as it will read better.

Reply:
Blame is on me for this one, felt it's unnecessary having its own method if it's used only once, and a bit weird to read:
```csharp
private void load()
{
    if (isAdmin())
        // post notification.
}

private bool isAdmin()
{
    ...
}
```

but not as much anymore.

Reply:
Moving to its own method would also allow for a `switch` statement to read quite well. the `IsWindows()` call could be nested within and `return false;` and it would still read well.